gpflow/models/training_mixins.py

```diff
@@ -0,0 +1,58 @@
+import abc
+import collections
+from typing import Callable, Optional, Tuple, TypeVar, Union
+
+
+import tensorflow as tf
+import numpy as np
+
+InputData = tf.Tensor
+OutputData = tf.Tensor
+RegressionData = Tuple[InputData, OutputData]
+Data = TypeVar("Data", RegressionData, InputData, OutputData)
```

Comment:
```suggestion
Data = Union[RegressionData, InputData, OutputData]
```

Reply:
`Data` isn't used in any places that require a `TypeVar`

Reply:
I believe TypeVar is the correct thing in this context.